subworkflows/local/deepvariant_error_polishing/main.nf

```diff
@@ -0,0 +1,312 @@
+    |- in case of multiple read files (therefore optional) merge all read files that belong to same assembly chunk (1..n)
+    |- index merged alignment files (1..n)
+    |- call variants with DeepVartiant (1..n)
+    |- filter variants (PASS + homozygous) (1..n)
+    |- create tabix index files  (1..n)
+    |- merge all variants (1)
+    |- create consensus sequence (1)
+*/
+
+workflow DVPOLISH {
+
+    take:
+    ch_assemblies // [ meta, assembly ]
+    ch_hifi       // [ meta, hifi ]
+    ch_meryl_hifi // [ meta, union.meryldb ] 
+
+    main:
+    
+    reads_plus_assembly_ch = combineByMetaKeys (
+            ch_hifi,
+            ch_assemblies,
+            keySet: ['id','sample'],
+            meta: 'rhs'
+        )
+
+    reads_plus_assembly_ch
+        // Add single_end for minimap module
+        .flatMap { meta, reads, assembly -> reads instanceof List ?
+            reads.collect{ [ meta + [ single_end: true ], it, assembly.pri_fasta ] }
+            : [ [ meta + [ single_end: true ], reads, assembly.pri_fasta ] ] }
+        .multiMap { meta, reads, assembly ->
+            reads_ch: [ meta + [ readID: reads.baseName ], reads ]
+            assembly_ch: [ meta, assembly ]
+        }
+        .set { input }
+
+    uniq_assembly_ch = getPrimaryAssembly(ch_assemblies)
+
+    assembly_plus_meryl_ch = combineByMetaKeys (
+            ch_meryl_hifi,
+            uniq_assembly_ch,
+            keySet: ['id','sample'],
+            meta: 'rhs'
+        )
+
+    // index assembly file(s)
+    SAMTOOLS_FAIDX (
+        uniq_assembly_ch,
+        [[],[]]
+    )    
+
+    // split assembly into smaller chunks, this step just creates bed files 
+    // that represent the assembly chunks, no sequence is split
+    DVPOLISH_CHUNKFA (
+        SAMTOOLS_FAIDX.out.fai
+    )
+
+    // create minimap2 index for assemblies
+    DVPOLISH_PBMM2_INDEX (
+        uniq_assembly_ch
+    )
+
+    // map reads with pbmm2 to complete assemblies (chunks are not used in that step)
+    DVPOLISH_PBMM2_ALIGN (
+        input.reads_ch,
+        input.assembly_ch
+    )
+
+    def path_closure = {meta, files -> files.collect(){[meta, it ]}}
+
+    combineByMetaKeys (
+        DVPOLISH_PBMM2_ALIGN.out.bam_bai,
+        DVPOLISH_CHUNKFA.out.bed.flatMap(path_closure),
+        keySet: ['sample','assembly'],
+        meta: 'rhs'
+    )
+    .multiMap { meta, bam, bai, bed ->
+        meta_bam_bai_ch:  [ meta + [ mergeID: bed.baseName ], bam, bai ]
+        meta_bed_ch:      [ meta + [ mergeID: bed.baseName ], bed ]
+        bed_ch:             bed
+    }
+    .set { alignment }
+    
+    // split bam files according to bed file chunks 
+    SAMTOOLS_VIEW (alignment.meta_bam_bai_ch,
+    [[],[]],                            
+    alignment.bed_ch)                   
+
+    // index the splitted bam files 
+    SAMTOOLS_INDEX_FILTER(SAMTOOLS_VIEW.out.bam)
+
+    SAMTOOLS_VIEW.out.bam
+    .groupTuple(by:0)
```

Comment:
If you happen to know what the `size:` of the tuples are this bit goes faster, otherwise it waits until all `SAMTOOLS_VIEW` are finished.

Reply:
I know the size, but how can I speed this up?